Runtime/Services/BacktraceSession.cs

```diff
@@ -0,0 +1,317 @@
+                return false;
+            }
+            // add to event attributes, attribute provider attributes
+            if (attributes == null)
+            {
+                attributes = new Dictionary<string, string>();
+            }
+            _attributeProvider.AddAttributes(attributes);
+
+            // validate if unique event attribute is available and
+            // prevent undefined attributes
+            if (attributes.TryGetValue(attributeName, out string attributeValue) == false || string.IsNullOrEmpty(attributeValue))
+            {
+                Debug.LogWarning("Attribute name is not available in attribute scope. Please define attribute to set unique event.");
+                return false;
+            }
+            var existingEvent = UniqueEvents.FirstOrDefault(m => m.Name == attributeName);
+            if (existingEvent != null)
+            {
+                UniqueEvents.Remove(existingEvent);
+            }
+
+            var @event = new UniqueEvent(attributeName, DateTimeHelper.Timestamp(), attributes);
+            UniqueEvents.AddLast(@event);
+            return true;
+        }
+
+        /// <summary>
+        /// Determine if Backtrace Session can add next event to store
+        /// </summary>
+        /// <param name="name">event name</param>
+        /// <returns>True if we're able to add. Otherwise false.</returns>
+        private bool ShouldProcessEvent(string name)
+        {
+            return !string.IsNullOrEmpty(name) && (_maximumNumberOfEventsInStore == 0 || (Count() + 1 <= _maximumNumberOfEventsInStore));
+        }
+
+
+        /// <summary>
+        /// Get number of events available right now in store
+        /// </summary>
+        /// <returns>number of events in store</returns>
+        public int Count()
+        {
+            return UniqueEvents.Count + SessionEvents.Count;
+        }
+
+        /// <summary>
+        /// Add session event to next Backtrace session request
+        /// </summary>
+        /// <param name="eventName">session event name</param>
+        public bool AddSessionEvent(string eventName, IDictionary<string, string> attributes = null)
+        {
+            if (!ShouldProcessEvent(eventName))
+            {
+                Debug.LogWarning("Skipping report: Reached store limit or event has empty name.");
+                return false;
+            }
+            var existingEvent = SessionEvents.FirstOrDefault(m => m.Name == eventName);
+            if (existingEvent != null)
+            {
+                SessionEvents.Remove(existingEvent);
+            }
+
+            var @event = new SessionEvent(eventName, DateTimeHelper.Timestamp(), attributes);
+            SessionEvents.AddLast(@event);
+            return true;
+        }
+
+        /// <summary>
+        /// Clean unique and session events stored in Backtrace Session
+        /// </summary>
+        public void CleanEvents()

```

Comment:
I think naming this `ClearEvents` would be a better name.

Reply:
Renamed. Thanks!